### PR TITLE
[GEOS-11306] Chace GetFeature total using AtomicLong for thread safety

### PR DESCRIPTION
**Pull request by jodygarnett**
[GEOS-11306](https://osgeo-org.atlassian.net/browse/GEOS-11306) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

This addresses a regression that was introduded with API change to avoid calculating total for formats that do not require such a value.

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [x] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [x] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by jodygarnett** [comment]
Since this is release blocker I am submitting as-is trusting to test coverage by existing tests.

I am trying to figure out if I can write a specific test to ensure value is not being called twice.

File: src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java [hunk 1]
```diff
@@ -711,10 +712,14 @@ private Supplier<BigInteger> updateTotalCount(
...
             return () -> BigInteger.valueOf(total);
         } else {
+            AtomicLong cache = new AtomicLong(Long.MIN_VALUE);
             return () -> {
                 try {
-                    long totalCount1 = getTotalCount(totalCountExecutors);
-                    return BigInteger.valueOf(totalCount1);
+                    if (cache.get() == Long.MIN_VALUE) {
+                        long totalCount1 = getTotalCount(totalCountExecutors);
+                        return BigInteger.valueOf(totalCount1);
```
Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java:720]: The cache should be updated here, no?

Comment (aaime) [inline src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java:720]: The cache should be updated here no?

Something  like this (untested):

```suggestion
                        long totalCount1 = getTotalCount(totalCountExecutors);
                        cache.set(totalCount1);
                        return BigInteger.valueOf(totalCount1);
```

Comment (jodygarnett) [reply src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java]: Yep, updated.

**Pull request merged**